Fluffyguy611/agile-project-team-A
# feat: added navigation bar

### 🛠 Changes being made

Added navigation bar to top of every webside, uses nunjacks to check if user is logged in and to check for admin rights

# How to use parent.html

1. Add on the top of website change "sample title" to your page title:

```
{% extends "./parent.html" %}
{% block content %}
<title>{% block title %}
"sample title"
{% endblock %}</title>
```

2. Start html inside <main></main>

3. End html file with:

```
{% endblock %}
```


CODE EXAMPLE:
```
{% extends "./parent.html" %}
{% block content %}
<title>{% block title %}Pizza{% endblock %}</title>
    
    <main class="container mt-4">
        <section id="jobRole" class="mt-4">
           {% if jobRole and jobRole.name.length > 0 %}
           <table class="table">
            <thead>
                <tr>
                    <th scope="col">Role Name</th>
                    <th scope="col">Role Description</th>
                    <th scope="col">SharePoint Link for Role Specification</th>
                </tr>
                <tbody>
                    <tr>
                        <td>{{ jobRole.name }}</td>
                        <td>{{ jobRole.description }} </td>
                        <td><a href="{{ jobRole.sharePointLink }}">View</a></td>
                    </tr>
                </tbody>
            </thead>
        </table>
           {% else %}
                Could not find job Role.
           {% endif %}
        </section>
    </main>

{% endblock %}
```



# HOW TO TEST
USE ../common/example_file.ts to change isLoggedIn and role to check for authorization 
```
export const example = {
    isLoggedIn: true,
    role: 'employee',
  };
 ```
 
 NAVIGATE BETWEEN /admin/add-job-roles and /job-roles/1 in html to check whenever "Currently at:" is changing. 
 
 ## PROOF OF WORKING
![Screenshot 2023-10-03 at 13 49 35](https://github.com/Fluffyguy611/agile-project-team-A/assets/93824235/c2141e49-5804-49b3-ac66-d1326dde6d40)
![Screenshot 2023-10-03 at 13 50 15](https://github.com/Fluffyguy611/agile-project-team-A/assets/93824235/cbe06bee-6a17-4b33-9a3e-cd0da500f0d4)
